jsonnet/prometheus-operator/admission-webhook.libsonnet

```diff
@@ -0,0 +1,123 @@
+local defaults = {
+  local defaults = self,
+  name: 'prometheus-operator-admission-webhook',
+  namespace: error 'must provide namespace',
+  version: error 'must provide version',
+  admissionWebhookImage: error 'must provide admissionWebhookImage',
+  port: 8443,
+  replicas: 2,
+  resources: {
+    limits: { cpu: '200m', memory: '200Mi' },
+    requests: { cpu: '50m', memory: '50Mi' },
+  },
+  commonLabels:: {
+    'app.kubernetes.io/name': defaults.name,
+    'app.kubernetes.io/version': defaults.version,
+  },
+  selectorLabels:: {
+    [labelName]: defaults.commonLabels[labelName]
+    for labelName in std.objectFields(defaults.commonLabels)
+    if !std.setMember(labelName, ['app.kubernetes.io/version'])
+  },
+};
+
+
+function(params) {
+  local aw = self,
+  config:: defaults + params,
+
+  serviceAccount: {
+    apiVersion: 'v1',
+    kind: 'ServiceAccount',
+    metadata: {
+      name: aw.config.name,
+      namespace: aw.config.namespace,
+      labels: aw.config.commonLabels,
+    },
```

Comment:
nit: for simplicity, we can extract metadata object into hidden `aw._metadata` and reference it in all the places as `metadata: aw._metadata`.

Like in this snippet: https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/components/blackbox-exporter.libsonnet#L97-L108